src/sentry/snuba/metrics/query_builder.py

```diff
@@ -212,6 +215,15 @@ def _parse_offset(self, query_params, paginator_kwargs):
                 raise InvalidParams("'cursor' is only supported in combination with 'orderBy'")
             return None
 
+    def _validate_series_limit(self, query_params):
+        if self.limit:
+            if (self.end - self.start).total_seconds() / self.rollup * self.limit > MAX_POINTS:
+                raise InvalidParams(
+                    f"Requested interval of {query_params.get('interval', '1h')} with statsPeriod of "
+                    f"{query_params.get('statsPeriod')} is too granular. Increase your interval or "
+                    f"decrease your statsPeriod"
```

Comment:
nit: Decreasing `page_size` is also an option, right?

Reply:
yeah good point